src/components/Promptbar/components/Prompt.tsx

```diff
@@ -48,12 +48,50 @@ export const PromptComponent = ({ item: prompt, level }: Props) => {
   const isSelected = selectedPromptId === prompt.id;
   const showModal = useAppSelector(PromptsSelectors.selectIsEditModalOpen);
 
+  const enabledFeatures = useAppSelector(
+    SettingsSelectors.selectEnabledFeatures,
+  );
+
   const [isDeleting, setIsDeleting] = useState(false);
   const [isRenaming, setIsRenaming] = useState(false);
   const [isShowMoveToModal, setIsShowMoveToModal] = useState(false);
+  const [isSharing, setIsSharing] = useState(false);
+  const { id: promptId, shares = [] } = prompt;
+  const isShared = shares.length > 0;
+  const isSharingEnabled = enabledFeatures.includes('prompts-sharing');
+  const showSharedIcon = isSharingEnabled && isShared && !isDeleting;
 
   const wrapperRef = useRef(null);
 
+  const handleCloseShareModal = useCallback(() => {
+    setIsSharing(false);
+  }, []);
+
+  const handleShared = useCallback(
+    (newShareId: string) => {
+      dispatch(
+        PromptsActions.updatePrompt({
+          promptId,
+          values: {
+            shares: [
+              ...shares,
+              {
```

Comment:
Are you sure that every time when you click share you will have new share id? I think it's working another way - you have for every entity(conversation, prompt) unique share id which will not regenerate when you clicking on share. 
Like it's working in figma - you can share single link to multiple people

Reply:
share link should be unique each time - because we have "This link is temporary and will be active for 3 days" text. I.e. if open share modal once and share with User1, wait e.g. 8 hours and open share modal, copy new link and share with User2 then second link for User2 will expire 8 later than first link for User1

Reply:
Do you really need this information for your conversation to being stored? I think just flag that it is shared will be enough, no?
Or it's something which is connected with other stuff out of this PR?

Reply:
@mikitabut, it's directly related to sharing functional: I store each unique shared Id with created date to be able to manage expiration and not allow to use it after 3 days

Reply:
So due to we don't have backend ready it's ok, but I think core api should take this role to manage expiration and other stuff, so maybe in future we don't need this

Reply:
@mikitabut we still need to store this information with dates related to specific conversation or prompt